File: src/api/java/lang/Object.java
package java.lang;

public class Object {
	public final Class getClass() {
		return VMManager.getClassNative(this);
	}
	
	public int hashCode() {
		return VMManager.getHashCodeNative(this);
	}
	
	public boolean equals(Object o) {
		return this == o;
	}
	
	public native String toString();
}

File: src/api/java/lang/Object.cpp
java::lang::String __Object::{overload:toString[]}(Object __this) {
	Class k = __this->__vptr->getClass(__this);
	std::ostringstream sout;
	sout << k->__vptr->getName(k) << '@' << std::hex << (uintptr_t)__this.raw();
	return new java::lang::__String(sout.str());	
}
